### PR TITLE
fix(kit): check if nuxt is restarting before updating templates

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

I've noticed a couple of times when restarting Nuxt, the templates are still being updated when the server restarts, and I get an error associated with the nuxt instance not being defined.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/kit/src/template.ts [hunk 1]
```diff
@@ -78,5 +78,5 @@ export function normalizeTemplate (template: NuxtTemplate<any> | string): Resolv
  * You can pass a filter within the options to selectively regenerate a subset of templates.
  */
 export function updateTemplates (options?: { filter?: (template: ResolvedNuxtTemplate<any>) => boolean }) {
-  return useNuxt().hooks.callHook('builder:generateApp', options)
+  return tryUseNuxt()?.hooks.callHook('builder:generateApp', options)
```
Comment (pi0) [inline packages/kit/src/template.ts:81]: Previously `callHook` always returned a Promise. While this is small, i think would be nice if we use `async/await` to make sure return type is promise (`?.` leads to `undefined`)

**Review by pi0 — APPROVED** [review]
Other than small comment LGTM 💯

File: packages/kit/src/template.ts [hunk 2]
```diff
@@ -78,5 +78,5 @@ export function normalizeTemplate (template: NuxtTemplate<any> | string): Resolv
  * You can pass a filter within the options to selectively regenerate a subset of templates.
  */
 export function updateTemplates (options?: { filter?: (template: ResolvedNuxtTemplate<any>) => boolean }) {
```
Comment (danielroe) [inline packages/kit/src/template.ts:80]:
```suggestion
export async function updateTemplates (options?: { filter?: (template: ResolvedNuxtTemplate<any>) => boolean }) {
```

File: packages/kit/src/template.ts (hunk 1 above)
Comment (danielroe) [inline packages/kit/src/template.ts:81]:
```suggestion
  return await tryUseNuxt()?.hooks.callHook('builder:generateApp', options)
```

**Pull request merged**